src/qtfirebaseadmob.cpp

```diff
@@ -1173,3 +1173,312 @@ void QtFirebaseAdMobInterstitial::show()
+{
+    if(state == QtFirebaseAdMobInterstitial::PresentationStateCoveringUI) {
+        if(!_visible) {
+            _visible = true;
+            emit visibleChanged();
+        }
+    }
+
+    if(state == QtFirebaseAdMobInterstitial::PresentationStateHidden) {
+        if(_visible) {
+            _visible = false;
+            emit visibleChanged();
+        }
+
+        setLoaded(false);
+        qDebug() << this << "::onPresentationStateChanged() loaded false";
+
+        // NOTE iOS necessities
+        /*#if defined(Q_OS_IOS)
+
+        setReady(false);
+        qDebug() << this << "::onPresentationStateChanged() ready false";
+
+        // Will be newed when init() is called
+        //delete _interstitial; // NOTE Crashes the app when used
+
+        // NOTE Auto re-initializing because of this: https://firebase.google.com/docs/admob/ios/interstitial#only_show_gadinterstitial_once
+        qDebug() << this << "::onPresentationStateChanged() re-initializing one-time use GADInterstitial";
+        _initTimer->start(500);
+
+        #endif*/
+
+        emit closed();
+    }
+}
+
+void QtFirebaseAdMobRewardedVideoAd::load()
+{
+    if(!_ready) {
+        qDebug() << this << "::load() not ready";
+        return;
+    }
+
+    if(_request == 0) {
+        qDebug() << this << "::load() no request data sat. Not loading";
+        return;
+    }
+
+    qDebug() << this << "::load() getting request data";
+    emit loading();
+    admob::AdRequest request = _request->asAdMobRequest();
+    firebase::FutureBase future = firebase::admob::rewarded_video::LoadAd( __adUnitIdByteArray.constData(), request);
+    qFirebase->addFuture(__QTFIREBASE_ID + ".rewardedvideoad.loaded",future);
+}
+
+void QtFirebaseAdMobRewardedVideoAd::show()
+{
+    if(!_ready) {
+        qDebug() << this << "::show() not ready - so not showing";
+        return;
+    }
+
+    if(!_loaded) {
+        qDebug() << this << "::show() not loaded - so not showing";
+        return;
+    }
+
+    _nativeUIElement = qFirebase->getNativeWindow();
+    firebase::FutureBase future = firebase::admob::rewarded_video::Show(static_cast<admob::AdParent>(_nativeUIElement));
+    qFirebase->waitForFutureCompletion(future); // TODO move or duplicate to QtFirebaseAdMob with admob::kAdMobError* checking? (Will save ALOT of cycles on errors)
+    if(future.error() != admob::kAdMobErrorNone) {
+        qDebug() << this << "::show ERROR code" << future.error() << "message" << future.error_message();
+        return;
+    }
+}
+
+void QtFirebaseAdMobRewardedVideoAd::OnRewarded(firebase::admob::rewarded_video::RewardItem reward)
+{
+    qDebug()<<QString("Rewarding user with %1")
+              .arg(QString::number(reward.amount));
+
+    //We could get reward type (which is identifier name in firebase console)
+    //but unfortunately any access of string object cause crash
+    //So just use amount only
+    //qDebug()<<this<<"award type empty:"<<reward.reward_type.empty();
```

Comment:
Sounds like a thing with the converstion - or the char pointer in firebase getting deleted before you get a chance to store it/use it? I've had a lot of problems with this. Read on:

Have you tried copying the actual char byte array (from the firebase) into a member of your class?

This is the reason I have stuff like these where I need to store a string (char**) from firebase:
```
QString _adUnitId;
QByteArray __adUnitIdByteArray; // <- this should store a copy of the chars/bytes in the array from the string you want to store from Firebase - this is important because QString byte array functions delete the contents after the function return - so we need to store it permanently on the class

// Then copy contents from the byte array to the QString - and use the QString (_adUnitId in this case) in the rest of the code
```

If it really is a bug in firebase - I've had luck reporting it here:
https://github.com/firebase/quickstart-cpp/issues?q=is%3Aopen+is%3Aissue